File: LearningManagementBackend/src/main/java/org/learn/learningmanagementbackend/dto/response/ScheduleExceptionResponse.java
package org.learn.learningmanagementbackend.dto.response;

import lombok.Getter;
import lombok.Setter;
import org.learn.learningmanagementbackend.enums.ApprovalStatus;
import org.learn.learningmanagementbackend.enums.ExceptionType;
import org.learn.learningmanagementbackend.enums.MakeupStatus;

import java.time.LocalDate;

@Getter
@Setter
public class ScheduleExceptionResponse {
    private Integer id;
    private Integer scheduleId;
    private String scheduleDetails; // Ví dụ: "INT101 - Thứ 2"
    private LocalDate exceptionDate;
    private String reason;
    private ExceptionType exceptionType;
    private LocalDate replacementDate;
    private ApprovalStatus approvalStatus;
    private String proofFileUrl;
    private Integer replacementStartPeriod;
    private Integer replacementEndPeriod;
    private String suggestedRoom;
    private String makeupNotes;
    private MakeupStatus makeupStatus;
    private Integer substituteTeacherId;
    private String substituteTeacherName;
    private String substituteContent;
    private ApprovalStatus substituteStatus;
    private Integer replacementRoomId;
    private String replacementRoomNumber;
    private String adminNote;       // Lý do từ chối
    private String teacherCode;     // Mã giảng viên gửi đề xuất
    private String teacherName;     // Tên giảng viên
    private String classCode;       // Mã lớp học phần
    private String courseName;      // Tên môn học
}

File: learning_management_app/lib/models/admin/schedule_exception_response.dart
class ScheduleExceptionResponse {
  final int? id;
  final int? scheduleId;
  final String? scheduleDetails;
  final String? exceptionDate;
  final String? reason;
  final String? exceptionType;
  final String? replacementDate;
  final String? approvalStatus;
  final String? proofFileUrl;
  final int? replacementStartPeriod;
  final int? replacementEndPeriod;
  final String? suggestedRoom;
  final String? makeupNotes;
  final String? makeupStatus;
  final int? substituteTeacherId;
  final String? substituteTeacherName;
  final String? substituteContent;
  final String? substituteStatus;
  final int? replacementRoomId;
  final String? replacementRoomNumber;
  final String? adminNote;
  final String? teacherCode;
  final String? teacherName;
  final String? classCode;
  final String? courseName;

  ScheduleExceptionResponse({
    this.id,
    this.scheduleId,
    this.scheduleDetails,
    this.exceptionDate,
    this.reason,
    this.exceptionType,
    this.replacementDate,
    this.approvalStatus,
    this.proofFileUrl,
    this.replacementStartPeriod,
    this.replacementEndPeriod,
    this.suggestedRoom,
    this.makeupNotes,
    this.makeupStatus,
    this.substituteTeacherId,
    this.substituteTeacherName,
    this.substituteContent,
    this.substituteStatus,
    this.replacementRoomId,
    this.replacementRoomNumber,
    this.adminNote,
    this.teacherCode,
    this.teacherName,
    this.classCode,
    this.courseName,
  });

  factory ScheduleExceptionResponse.fromJson(Map<String, dynamic> json) {
    return ScheduleExceptionResponse(
      id: json['id'],
      scheduleId: json['scheduleId'],
      scheduleDetails: json['scheduleDetails'],
      exceptionDate: json['exceptionDate'],
      reason: json['reason'],
      exceptionType: json['exceptionType'],
      replacementDate: json['replacementDate'],
      approvalStatus: json['approvalStatus'],
      proofFileUrl: json['proofFileUrl'],
      replacementStartPeriod: json['replacementStartPeriod'],
      replacementEndPeriod: json['replacementEndPeriod'],
      suggestedRoom: json['suggestedRoom'],
      makeupNotes: json['makeupNotes'],
      makeupStatus: json['makeupStatus'],
      substituteTeacherId: json['substituteTeacherId'],
      substituteTeacherName: json['substituteTeacherName'],
      substituteContent: json['substituteContent'],
      substituteStatus: json['substituteStatus'],
      replacementRoomId: json['replacementRoomId'],
      replacementRoomNumber: json['replacementRoomNumber'],
      adminNote: json['adminNote'],
      teacherCode: json['teacherCode'],
      teacherName: json['teacherName'],
      classCode: json['classCode'],
      courseName: json['courseName'],
    );
  }
}
